include/cppkafka/header_list.h

```diff
@@ -146,6 +146,12 @@ class HeaderList {
      */
     rd_kafka_headers_t* get_handle() const;
     
+    /**
+     * \brief Clone the underlying header list handle.
+     * \return The handle.
+     */
+    rd_kafka_headers_t* clone_handle() const;
```

Comment:
What is the reason for this member function? Isn't it the equivalent of copying the `HeaderList` via copy constructor?

Reply:
The reason is explained above in your comment about the `ClonablePtr::clone()`. You can't use the copy constructor if you build a HeaderList<A> from a HeaderList<B>. So this was a way of doing this transformation instead of working with the raw handle. But in the end, since cloning does not guaratee a copy, I will remove it.